File: python/priorityQueue/disJointSetTree.py
class DisJointSetTree:
    def __init__(self):
        self.parent = [None] * 9
        self.rank = [None] * 9

    def toString(self):
        return ','.join([str(p) for p in self.parent if p is not None])

    def toStringRank(self):
        return ','.join([str(p) for p in self.rank if p is not None])

    def makeSet(self, i):
        self.parent[i] = i
        self.rank[i] = 0

    def find(self, i):
        if i < 0 or i >= len(self.parent):
            raise IndexError
        
        while i != self.parent[i]:
            i = self.parent[i]
        
        return self.parent[i]

    def union(self, i, j):
        i_id = self.find(i)
        j_id = self.find(j)

        if self.rank[i_id] > self.rank[j_id]:
            self.parent[j_id] = i_id
        else:
            self.parent[i_id] = j_id
            if self.rank[i_id] == self.rank[j_id]:
                self.rank[j_id] += 1

File: java/src/priorityQueue/DisJointSetTree.java
package priorityQueue;

public class DisJointSetTree {
    private int[] parent = new int[6];

    private int[] rank = new int[6];

    @Override
    public String toString() {
        StringBuffer sb = new StringBuffer();
        for(int p : parent) {
            sb.append(p);
            sb.append(",");
        }

        return sb.toString();
    }

    public void makeSet(int i) {
        if (i < 0 || i >= this.parent.length)
            throw new IndexOutOfBoundsException();
        this.parent[i] = i;
    }

    public int find(int i) {
        if (i < 0 || i >= this.parent.length)
            throw new IndexOutOfBoundsException();

        while (i != this.parent[i])
            i = this.parent[i];

        return this.parent[i];
    }

    public void union(int i , int j) {
        int i_id = this.find(i);
        int j_id = this.find(j);

        if (this.rank[i_id] > this.rank[j_id]) {
            this.parent[j_id] = i_id;
        } else {
            this.parent[i_id] = j_id;
            if (this.rank[i_id] == this.rank[j_id])
                this.rank[j_id] += 1;
        }
    }
}
